download/download.go

```diff
@@ -97,30 +101,62 @@ func (d *Downloader) ClearCache() {
 
 // Start tells the Downloader to begin downloading bundles.
 func (d *Downloader) Start(ctx context.Context) {
-	go d.loop()
+	go d.doStart(ctx)
 }
 
-// Stop tells the Downloader to stop begin downloading bundles.
+func (d *Downloader) doStart(ctx context.Context) {
+	ctx, cancel := context.WithCancel(context.Background())
+
+	d.wg.Add(1)
+	go d.loop(ctx)
+
+	for {
+		select {
+		case done := <-d.stop:
+			cancel()
+			d.wg.Wait()
+			done <- struct{}{}
+			return
+		}
+	}
+}
+
+// Stop tells the Downloader to stop downloading bundles.
 func (d *Downloader) Stop(ctx context.Context) {
```

Comment:
[nit] The context isn't used, so we might as well do
```suggestion
func (d *Downloader) Stop(context.Context) {
```

Reply:
Done.